samza-core/src/main/java/org/apache/samza/runtime/ApplicationRunnerMain.java

```diff
@@ -34,19 +34,48 @@
  */
 public class ApplicationRunnerMain {
   // TODO: have the app configs consolidated in one place
-  private static final String STREAM_APPLICATION_CLASS_CONFIG = "app.class";
+  public static final String STREAM_APPLICATION_CLASS_CONFIG = "app.class";
+
+  public static class ApplicationRunnerCommandLine extends CommandLine {
+    public OptionSpec operationOpt =
+        parser().accepts("operation", "The operation to perform; run, status, kill.")
+            .withRequiredArg()
+            .ofType(String.class)
+            .describedAs("operation=run")
+            .defaultsTo("run");
+
+    public ApplicationRunnerOperation getOperation(OptionSet options) {
+      String rawOp = options.valueOf(operationOpt).toString();
+      return ApplicationRunnerOperation.fromString(rawOp);
+    }
+  }
 
   public static void main(String[] args) throws Exception {
-    CommandLine cmdLine = new CommandLine();
+    ApplicationRunnerCommandLine cmdLine = new ApplicationRunnerCommandLine();
     OptionSet options = cmdLine.parser().parse(args);
     Config config = cmdLine.loadConfig(options);
+    ApplicationRunnerOperation op = cmdLine.getOperation(options);
 
     if (config.containsKey(STREAM_APPLICATION_CLASS_CONFIG)) {
       ApplicationRunner runner = ApplicationRunner.fromConfig(config);
-      StreamApplication app = (StreamApplication) Class.forName(config.get(STREAM_APPLICATION_CLASS_CONFIG)).newInstance();
-      runner.run(app);
+      StreamApplication app =
+          (StreamApplication) Class.forName(config.get(STREAM_APPLICATION_CLASS_CONFIG)).newInstance();
+      switch (op) {
+        case RUN:
+          runner.run(app);
+          break;
+        case KILL:
+          runner.kill(app);
+          break;
+        case STATUS:
+          System.out.println(runner.status(app));
```

Comment:
The invoking method. 

At LI, the control script will see this output and interpret it. 

For the record, I don't love this approach, but the only other idea I had was to encode meanings into a set of exit codes.